packages/chat-stateful-client/src/ChatContext.ts

```diff
@@ -33,68 +34,66 @@ export class ChatContext {
     latestErrors: {} as ChatErrors
   };
   private _batchMode = false;
+  private _logger: AzureLogger;
   private _emitter: EventEmitter;
   private typingIndicatorInterval: number | undefined = undefined;
 
   constructor(maxListeners?: number) {
+    this._logger = createClientLogger('communication-react:chat-context');
     this._emitter = new EventEmitter();
     if (maxListeners) {
       this._emitter.setMaxListeners(maxListeners);
     }
   }
 
+  public getState(): ChatClientState {
+    return this._state;
+  }
+
   public setState(state: ChatClientState): void {
     this._state = state;
     if (!this._batchMode) {
       this._emitter.emit('stateChanged', this._state);
     }
   }
 
-  public getState(): ChatClientState {
-    return this._state;
-  }
-
-  public modifyState(modifier: ChatStateModifier): void {
-    this.batch(() => {
-      this.setState(
-        produce(this._state, (draft: ChatClientState) => {
-          modifier(draft);
-        })
-      );
+  public modifyState(modifier: (draft: ChatClientState) => void): void {
+    this._state = produce(this._state, modifier, (patches: Patch[]) => {
+      if (getLogLevel() === 'verbose') {
```

Comment:
In my mind, it sounds like verbose < info < warnings

So my understanding here, we wanna still keep this one as verbose level but log the information using logger.info just because of the need of console output, but not to log anything when user set it as info level. Is this a correct understanding?

Reply:
Yes. This log messages are quite verbose, but I can't just `log.verbose` because they don't show up on console by default. :(

By contrast, @azure/communication-calling spews a lot of logs on `log.info`, which I dislike quite a bit.